Update Interoperability doc to explain how to use Ballerina complex types in Java (#1739)
ballerina-platform/ballerina-dev-website: swan-lake/learn/calling-java-code-from-ballerina.md

byte | byte, short, char, int, long, float, double | Widening conversion from byte -> short, char, int, long, float, double
int | byte, char, short, int, long | Narrowing conversion when int -> byte, char, short, and int
float | byte, char, short, int, long, float, double | Narrowing conversion when float -> byte, char, short, int, long, float
string | io.ballerina.runtime.api.values.BString |
xml | io.ballerina.runtime.api.values.BXml |
array | io.ballerina.runtime.api.values.BArray |
tuple | io.ballerina.runtime.api.values.BArray |
map | io.ballerina.runtime.api.values.BMap |
table | io.ballerina.runtime.api.values.BTable | 
stream | io.ballerina.runtime.api.values.BStream |
object | io.ballerina.runtime.api.values.BObject |
future | io.ballerina.runtime.api.values.BFuture |
function | io.ballerina.runtime.api.values.BFunctionPointer |
typedesc | io.ballerina.runtime.api.values.BTypedesc |
error | io.ballerina.runtime.api.values.BError |

#### Using Ballerina Arrays and Maps in Java

> There is no direct mapping between Ballerina arrays and maps to primitive Java arrays and maps. In order to facilitate the use of Ballerina arrays and maps in Java, `ballerina-runtime` libraries have to be added as a dependency to the java project and the relevant classes need to be imported from the `ballerina-runtime` library. You can find all released versions of the `ballerina-runtime` library [here](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/) and add the latest version of it as a dependency by adding following lines to `build.gradle` file.

Reviewer: ```suggestion
There is no direct mapping between Ballerina arrays and maps to primitive Java arrays and maps. In order to facilitate the use of Ballerina arrays and maps in Java, the `ballerina-runtime` libraries have to be added as a dependency to the Java project and the relevant classes need to be imported from the `ballerina-runtime` library. You can find all the released versions of the [`ballerina-runtime` library](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/), and add the latest version of it as a dependency by adding the following to the `build.gradle` file.
```

> There is no direct mapping between Ballerina arrays and maps to primitive Java arrays and maps. In order to facilitate the use of Ballerina arrays and maps in Java, `ballerina-runtime` libraries have to be added as a dependency to the java project and the relevant classes need to be imported from the `ballerina-runtime` library. You can find all released versions of the `ballerina-runtime` library [here](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/) and add the latest version of it as a dependency by adding following lines to `build.gradle` file.

Reviewer 2: ```suggestion
There is no direct mapping between Ballerina arrays and maps to primitive Java arrays and maps. In order to facilitate the use of Ballerina arrays and maps in Java, `ballerina-runtime` libraries have to be added as a dependency to the java project and the relevant classes need to be imported from the `ballerina-runtime` library. You can find all released versions of the `ballerina-runtime` library [here](https://maven.wso2.org/nexus/content/repositories/releases/org/ballerinalang/ballerina-runtime/). The latest version of the dependency can be added to gradle using the following:
```

```groovy
repositories {
   // Use WSO2's Nexus repository manager for resolving dependencies.
   maven {
       url = 'https://maven.wso2.org/nexus/content/repositories/releases/'
   }
}

dependencies {
   // Add ballerina-runtime:2.0.0-Preview8 as dependency.
   implementation 'org.ballerinalang:ballerina-runtime:+'
}
```

##### Using Ballerina Arrays in Java

> To use Ballerina arrays in Java, `BArray` interface has been provided. Following example illustrates how to write Java intreop code that uses Ballerina arrays.

Reviewer: ```suggestion
To use Ballerina arrays in Java, the `BArray` interface has to be used. The example below illustrates how to write Java intreop code that uses Ballerina arrays.
```

```java
import io.ballerina.runtime.api.values.BArray;

public class ArrayReverse {
   public static BArray arrayReverse(BArray arr) {
       long len = arr.size();
       for(long i = len - 1, j = 0; j < len/2; i--, j++) {
           Object temp = arr.get(j);
           arr.add(j, arr.get(i));
           arr.add(i, temp);
       }
       return arr;
   }
}
```
Associated Ballerina code:
```ballerina
import ballerina/io;
import ballerina/jballerina.java;
 
public function main() {
   int[] a = [1,2,3,4,5];
   int[] b = arrayReverse(a);
   io:println(b);
}
 
function arrayReverse(int[] arr) returns int[] = @java:Method {
   'class: "javalibs.app.ArrayReverse"
} external;
```

##### Using Ballerina Maps in Java

> To use Ballerina maps in Java, `BMap` interface has been provided. Following example illustrates how to write Java intreop code that uses Ballerina maps.

Reviewer: ```suggestion
To use Ballerina maps in Java, the `BMap` interface has to be used. The example below illustrates how to write Java intreop code that uses Ballerina maps.
```
